app/models/user.rb

```diff
@@ -34,6 +34,8 @@ class User < ActiveRecord::Base
   validates :email, uniqueness: true, allow_blank: true,
                     format: {with: /\A[^@\s]+@[^@\s]+\Z/}
 
+  validates :email, fake_email: true, :if => lambda { Rails.env.staging? || Rails.env.production? }
```

Comment:
I'm always wary of code that doesn't run in certain environments. In this case, we'll never catch it running in development so hard to be sure things haven't broken before pushing to staging or live etc.

I checked the blacklist. Emails ending with 'example.org' or 'goodcity.hk' would pass for testing so why don't we use those and remove the restriction.

Reply:
@steveyken We're actually seeing issues with the fake_email_validator (check slack), so that makes it a very fair point : )

Reply:
@steveyken  @patrixr , also we can fork this gem, and add our own blacklist.

cc - @namrataukirde

Reply:
Ideally it should only run only on production and for Staging and PROD we should have separate mailing services, so that testing doesn's spoil the reputation for PROD mailing service.
Thoughts ?
cc - @steveyken  @patrixr  @namrataukirde